From 4b1d29d2831717b66667cf901c1124ed77fef1a3 Mon Sep 17 00:00:00 2001
From: Jingchang Lu <jingchang.lu@freescale.com>
Date: Fri, 2 May 2014 09:34:05 +0800
Subject: [PATCH 095/255] arm: dts: ls1021a-40b: changes for pcie node

make root "#address-cells" to 2, and make changes on
pcie interrupts propertis accordingly.

Signed-off-by: Jingchang Lu <jingchang.lu@freescale.com>
[Kevin: Original patch taken from
LS1021A-SDK-V1.1-ARM-SOURCE-20140815-yocto.iso]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm/boot/dts/ls1021a-40b.dtsi | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/arch/arm/boot/dts/ls1021a-40b.dtsi b/arch/arm/boot/dts/ls1021a-40b.dtsi
index bda7377..d595f87 100644
--- a/arch/arm/boot/dts/ls1021a-40b.dtsi
+++ b/arch/arm/boot/dts/ls1021a-40b.dtsi
@@ -13,6 +13,7 @@
 / {
 	compatible = "fsl,ls1021a";
 	interrupt-parent = <&gic>;
+	#address-cells = <2>;
 
 	aliases {
 		serial0 = &lpuart0;
@@ -31,7 +32,7 @@
 	};
 
 	memory {
-		reg = <0x80000000 0x20000000>;
+		reg = <0x0 0x80000000 0x20000000>;
 	};
 
 	cpus {
@@ -69,7 +70,7 @@
 		#size-cells = <1>;
 		compatible = "simple-bus";
 		interrupt-parent = <&gic>;
-		ranges = <0x0 0x00 0x00000000 0x10000000>;
+		ranges;
 
 		gic: interrupt-controller@1400000 {
 			compatible = "arm,cortex-a15-gic";
@@ -606,7 +607,7 @@
 		pcie@3400000 {
 			compatible = "fsl,pcie", "snps,dw-pcie", "fsl,ls1021a-pcie";
 			reg = <0x00 0x3400000 0x10000>;
-			interrupts = <0 177 0>, <0 179 0>, <0 181 0>, <0 183 0>;
+			interrupts = <0 177 0x04>, <0 179 0x04>, <0 181 0x04>, <0 183 0x04>;
 			interrupt-names = "intr", "msi", "pme", "cfg-err";
 			#address-cells = <3>;
 			#size-cells = <2>;
@@ -617,10 +618,10 @@
 				  0x81000000 0x0 0x00000000 0x40 0x10000000 0x0 0x00010000 /* downstream I/O */
 				  0x82000000 0x0 0x00000000 0x41 0x00000000 0x1 0x00000000>; /* non-prefetchable memory */
 			interrupt-map-mask = <0 0 0 7>;
-			interrupt-map = <0000 0 0 1 &gic 0 90  0>,
-					<0000 0 0 2 &gic 0 188 0>,
-					<0000 0 0 3 &gic 0 190 0>,
-					<0000 0 0 4 &gic 0 192 0>;
+			interrupt-map = <0000 0 0 1 &gic 0 91  0x04>,
+					<0000 0 0 2 &gic 0 188 0x04>,
+					<0000 0 0 3 &gic 0 190 0x04>,
+					<0000 0 0 4 &gic 0 192 0x04>;
 			};
 
 		pcie@3500000 {
@@ -631,7 +632,7 @@
 			#address-cells = <3>;
 			#size-cells = <2>;
 			device_type = "pci";
-			num-lanes = <4>;
+			num-lanes = <2>;
 			bus-range = <0x0 0xff>;
 			ranges = <0x82000000 0x0 0x00000000 0x48 0x00000000 0x0 0x10000000 /* configuration space */
 				  0x81000000 0x0 0x00000000 0x48 0x10000000 0x0 0x00010000 /* downstream I/O */
-- 
2.0.2

